userspace/libsinsp/test/sinsp_utils.ut.cpp

```diff
@@ -0,0 +1,123 @@
+	EXPECT_EQ("/dir/term", std::string(target));
+
+	path = "/dir/../term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/term", std::string(target));
+
+	path = "/dir/dir/../term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/term", std::string(target));
+
+	path = "/dir/dir/../../term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/term", std::string(target));
+
+	path = "/dir/term/..";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir", std::string(target));
+
+	path = "/dir/./term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/term", std::string(target));
+
+	path = "/dir/././term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/term", std::string(target));
+
+	path = "/dir/term/.";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/term", std::string(target));
+
+	path = "/dir/.../term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/.../term", std::string(target));
+
+	path = "/dir/term/...";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/term/...", std::string(target));
+
+	path = "/dir/term/....";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/term/....", std::string(target));
+
+	path = "/dir/..../term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/..../term", std::string(target));
+
+	path = "/dir/dir../term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/dir../term", std::string(target));
+
+	path = "/dir/dir./term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/dir./term", std::string(target));
+
+	path = "/dir/..dir/term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/..dir/term", std::string(target));
+
+	path = "/dir/.dir/term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/.dir/term", std::string(target));
+
+	path = ".dir";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ(".dir", std::string(target));
+
+	path = "./";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("", std::string(target));
+
+	path = "./.";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("", std::string(target));
+
+	path = ".";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("", std::string(target));
+
+	path = "../";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("", std::string(target));
+
+	path = "../..";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("", std::string(target));
+
+	path = "..";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("", std::string(target));
```

Comment:
Those are awesome, thank you so much! :rocket: ... the only other case that isn't tested is when the two arguments are not the same. If we switch to `concatenate_paths` we can easily add a couple of tests for that.

Reply:
Oh, good suggestion! I can add those. 

I just pushed a few more tests with double `/` suggested by @gentooise